### ISSUE TITLE
Work on pkg-config support

### ISSUE THREAD

**Issue by jeffreykegler** [jeffreykegler/libmarpa#87]
Continuing work on pkg-config.

**Comment by jeffreykegler** [comment]
@dabrahams I am opening this to record notes and commentary on what I am doing about pkg-config.

**Comment by jeffreykegler** [comment]
I am changing the prefix variable to a relative path.  This is inconvenient is your ultimate destination is indeed "/usr/local", but this repo is intended more generally -- it may be used by a firmware developer, for example.  To avoid accidents on the part of downstream developers, I want to keep absolute paths out of this package's pkg-config.  This inconveniences more users than it helps, but I want the developer/user to have confidence that I've kept potential writes to absolute locations to a minimum.  A perspective on these tradeoffs is [here](https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/).

**Comment by jeffreykegler** [comment]
Work completed with https://github.com/jeffreykegler/libmarpa/commit/11089b107d46acdf93a41baee37e530f2401a410.  Absent feedback to the contrary, I will close this issue in a couple of days.

**Comment by dabrahams** [comment]
Your change to a relative path may have some kind of theoretical purity, but it's inconsistent with all of the `.pc` files installed on my system and it causes pkg-config to not work at all. You appear to have misinterpreted the advice in that article, which is badly worded: it seems to actually mean that all the paths other than `prefix` should transitively incorporate `prefix` in some form so that the location can be customized.  None of the examples in that article actually make `prefix` itself a relative path.  Testing variations like this one is always a good idea

**Comment by jeffreykegler** [comment]
Currently all of libmarpa's direct users use relative paths only.  The absolute path presents a danger to them -- accidental writing of absolute paths could be annoyance, or even a disaster.  Even if the majority of libmarpa's direct users used specific absolute paths, I would go with the safety of the minority.

I dislike to "dig in my heels" on a topic, but I toyed with the idea of absolute paths and could not get over a strong visceral discomfort with the idea, so to be honest I'd be hard to move on this topic.  My concern is not out of "theoretical purity", though it might be called over-cautious or paranoid.

I'll "reset the clock" and wait a couple of days to allow further discussion.  After that, absent any new angles on this issue, I will close this issue.

**Comment by dabrahams** [comment]
- Libmarpa's current set of direct users is infinitesimal compared to what it should be, if the technology works as advertised.
- pkgconfig support is one of those things you need in order to make a library like this widely consumable across a variety of different platforms, and thus could be considered a requirement for adoption.
- the whole point of pkg-config is to encapsulate details of how/where a thing is actually installed.  Not having an absolute path in the generated file makes it practically useless.
- the pkg-config file only gets generated in the **install** step of the build, which is designed to put things into their final locations.  In fact CMake will reject a relative installation prefix, because it's really never the right thing to do.

In other words, IIUC if you don't “do it my way,” you may as well drop the whole idea of pkgconfig support, and everybody who packages libmarpa for installation will have to manually go through the process of trying to create the right `.pc` file. 🤷 it's up to you of course.

**Comment by jeffreykegler** [comment]
A digression: 

The technology to the best of my knowledge *does* work as advertised.  Alas, in the modern parsing world, it's reasonable for folks to not take claims at face value.  Many people have advised me, when I don't really have evidence of a claim, to make it anyway.  And these people are apparently well-meaning:  "Fake it til you make it" seems to be accepted practice in the world of parsing.

Libmarpa has many direct and indirect users, as CPAN and MetaCPAN show.  (Although I agree with you the numbers are not what they "should be".)  This gives me some confidence that my claims are met, and in a way that is practical.

But, as the pioneers of parsing knew, and many current practitioners have forgotten, number of users and applications is not enough.  Even the most careful testing is only partially effective when it comes something as complex, and with as many cases, as a parser.  Marpa has the necessary (and traditionally required) proof apparatus.  This assures me that, even when a problem with my claims is encountered in implementation, it is an artifact of the implementation, and not a necessary consequence of the approach.

Of course, like everyone, I am fallible, so that when I speak of certainty, I must be taken to mean a very high level of confidence.  But with Marpa, every reasonable measure has been taken to justify that level of confidence.

**Comment by jeffreykegler** [comment]
OK, back to the issue:

I read https://github.com/jeffreykegler/libmarpa/issues/87#issuecomment-1100704199 as saying that as implemented (since the direct libmarpa user is almost always required to edit the pkg-config file) my pkg-config support may as well be removed.  To avoid misleading users, and to minimize clutter, I will do so.

I will remove the pkg-config support.  Once that is done, I will re-propose this issue for closing.

**Comment by dabrahams** [comment]
Actually, it would be marginally easier for me if you leave pkg-config support in place, and I update the homebrew formula to patch the generated .pc file.

**Comment by jeffreykegler** [comment]
OK, thanks!  Per https://github.com/jeffreykegler/libmarpa/issues/87#issuecomment-1101934722, I am marking this "about to be closed".  Absent feedback to the contrary, I will close this in a couple of days.

**Comment by jeffreykegler** [comment]
At some point, I may do further research and reconsider the handling of pkgconfig directories.  For now, absent further comments, I will close this issue is a couple of days.
